src/brevitas_examples/imagenet_classification/ptq/learned_round_utils.py

```diff
@@ -26,195 +26,200 @@
+
+        self["inputs"].append(input_batch)
+
+    def store_output(self, output) -> None:
+        if self.batch_dim is not None:
+            output.rename_(None)
+            output = output.transpose(0, self.batch_dim)
+
+        self["output"].append(output)
+
+    def initialize_cache(self) -> None:
+        self["inputs"] = []
+        self["output"] = []
+
+    def clear_cache(self) -> None:
+        del self["inputs"]
+        del self["output"]
+        self["inputs"] = []
+        self["output"] = []
+
+    def reset_cache(self) -> None:
+        del self["inputs"]
+        del self["output"]
+        self["inputs"] = []
+        self["output"] = []
+
+    def sample_batch(self, indices: torch.Tensor) -> Union[Any, torch.Tensor]:
+        if isinstance(self["inputs"], list):
+            self["inputs"] = torch.cat(self["inputs"], dim=self.batch_dim)
+        if isinstance(self["output"], list):
+            self["output"] = torch.cat(self["output"], dim=self.batch_dim)
+
+        return self["inputs"][indices], self["output"][indices]
+
+    def cache_to_dataset(self) -> Dataset:
+        raise NotImplementedError("This method is still not available for CNNs.")
+
+    def collate_fn(self, batch: Any) -> Any:
+        raise NotImplementedError("This method is still not available for CNNs.")
+
+    def __len__(self):
+        return (
+            len(self["inputs"])
+            if isinstance(self["inputs"], list) else self["inputs"].shape[self.batch_dim])
+
+
+def cnn_forward(model: nn.Module, inputs: Any) -> None:
     device = next(model.parameters()).device
-    data_saver = DataSaverHook(store_output=store_out)
-    handle = module.register_forward_hook(data_saver)
-    cached = [[], []]
-    with torch.no_grad():
-        for img, t in dataloader:
-            try:
-                _ = model(img.to(device))
-            except StopFwdException:
-                pass
-            if store_inp:
-                if keep_gpu:
-                    cached[0].append(data_saver.input_store.detach())
-                else:
-                    cached[0].append(data_saver.input_store.detach().cpu())
-            if store_out:
-                if keep_gpu:
-                    cached[1].append(data_saver.output_store.detach())
-                else:
-                    cached[1].append(data_saver.output_store.detach().cpu())
-    if store_inp:
-        cached[0] = torch.cat([x for x in cached[0]])
-    if store_out:
-        cached[1] = torch.cat([x for x in cached[1]])
-    handle.remove()
-    if disable_quant:
-        disable_quant_class.enable_act_quantization(model, False)
-        disable_quant_class.enable_param_quantization(model, False)
-        restore_return_quant_tensor(model, return_quant_tensor_state)
-    return cached
+    img, _ = inputs
+    img = send_to_device(img, device)
+    model(img)
+
+
+def cnn_block_forward(block: nn.Module, inputs: Any) -> torch.Tensor:
```

Comment:
There are minimal differences between this and the previous func. Can we have only one?

Reply:
What about this?